test/aws/rails/middleware/elastic_beanstalk_worker_listener_test.rb

```diff
@@ -0,0 +1,95 @@
+require 'test_helper'
+require_relative 'elastic_beanstalk_job'
+require_relative 'elastic_beanstalk_periodic_task'
+
+module Aws
+  module Rails
+    describe ElasticBeanstalkWorkerListener do
+      # Simple mock Rack app that always returns 200
+      let(:mock_rack_app) { ->(_) { [200, { 'Content-Type' => 'text/plain' }, ['OK']] } }
+
+      let(:logger) { double(error: nil, debug: nil, warn: nil) }
+
+      it 'passes request through if user-agent is not SQS Daemon' do
+        mock_rack_env = create_mock_env('127.0.0.1', 'not-aws-sqsd')
+
+        test_middleware = ElasticBeanstalkWorkerListener.new(mock_rack_app)
+        response = test_middleware.call(mock_rack_env)
+
+        expect(response[0]).to eq(200)
+        expect(response[2]).to eq(['OK'])
+      end
+
+      it 'returns forbidden when called from untrusted source' do
+        mock_rack_env = create_mock_env('1.2.3.4', 'aws-sqsd/1.1')
+
+        test_middleware = ElasticBeanstalkWorkerListener.new(mock_rack_app)
+        response = test_middleware.call(mock_rack_env)
+
+        expect(response[0]).to eq(403)
+      end
+
+      it 'successfully invokes job when passed through request body' do
+        # Stub execute call to avoid invoking Active Job callbacks
+        allow(ActiveJob::Base).to receive(:execute).and_return(nil)
+        mock_rack_env = create_mock_env('::1', 'aws-sqsd/1.1')
+
+        test_middleware = ElasticBeanstalkWorkerListener.new(mock_rack_app)
+        response = test_middleware.call(mock_rack_env)
+
+        expect(response[0]).to eq(200)
+        expect(response[1]['Content-Type']).to eq('text/plain')
+        expect(response[2]).to eq(['Successfully ran job ElasticBeanstalkJob.'])
+      end
+
+      it 'returns internal server error if job name cannot be resolved' do
+        # Stub execute call to avoid invoking Active Job callbacks
+        # Local testing indicates this failure results in a NameError
+        allow(ActiveJob::Base).to receive(:execute).and_raise(NameError)
+        mock_rack_env = create_mock_env('::1', 'aws-sqsd/1.1')
+
+        test_middleware = ElasticBeanstalkWorkerListener.new(mock_rack_app)
+        response = test_middleware.call(mock_rack_env)
+
+        expect(response[0]).to eq(500)
+      end
+
+      it 'successfully invokes periodic task when passed through custom header' do
+        mock_rack_env = create_mock_env('127.0.0.1', 'aws-sqsd/1.1', true)
+
+        test_middleware = ElasticBeanstalkWorkerListener.new(mock_rack_app)
+        response = test_middleware.call(mock_rack_env)
+
+        expect(response[0]).to eq(200)
+        expect(response[1]['Content-Type']).to eq('text/plain')
+        expect(response[2]).to eq(['Successfully ran periodic task ElasticBeanstalkPeriodicTask.'])
```

Comment:
can we add an expectation that the `perform_now` method gets called?  Something like `expect_any_instance_of(ElasticBeanstalkPeriodicTask).to receive(:perform_now)`

Reply:
Didn't realize you could do that! Naturally that's what I wanted to validate, I'll add the assertion.